org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm

```diff
@@ -1,13 +1,58 @@
+from abc import ABC
+
+class Signature(ABC):
+    def __init__(self, sig_name: str = "Signature", name: str = None) -> None:
+        if name is None:
+            self._name = self.__class__.__name__
+        else:
+            self._name = name
+        self._sig_name = sig_name
+
+    def __repr__(self):
+        return f"{self._sig_name} {self._name}"
+
+    def __key(self):
+        return self._sig_name + self._name
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, other):
+        if isinstance(other, Signature):
+            return self.__key() == other.__key()
+        return False
+
+
+# Model "one sig" as singleton classes
+class Singleton(type(Signature)):
+_instances = {}
+
+def __call__(cls, *args, **kwargs):
+if cls not in cls._instances:
+cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+return cls._instances[cls]
+
 # --- Signatures ---
 
 #foreach($label in $basicSigLabels)
 class $label(Signature):
     def __init__(self, name: str = None) -> None:
         super().__init__("${label}", name)
 
+#end
+
+#foreach($label in $oneSigLabels)
+class $label(Signature, metaclass=Singleton):
+    def __init__(self):
+        self._abbrev = '${label.charAt(0)}'
```

Comment:
Just curious, why do we need this?

Reply:
For user input purposes later, same as in the farmer manual translation.